apps/retrospective/services/sprint.go

```diff
@@ -127,19 +127,129 @@ func (service SprintService) Get(sprintID string) (*retroSerializers.Sprint, int
 	}
 
 	return &sprint, http.StatusOK, nil
 }
 
+// GetSprintSummary ...
+func (service SprintService) GetSprintSummary(sprintID string, retroID uint) (summary retroSerializers.SprintSummary, status int, err error) {
+	db := service.DB
+
+	var sprint retroModels.Sprint
+
+	err = db.Model(&retroModels.Sprint{}).Where("id = ?", sprintID).Preload("Retrospective").First(&sprint).Error
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return summary, http.StatusNotFound, errors.New("sprint not found")
+		}
+		utils.LogToSentry(err)
+		return summary, http.StatusInternalServerError, errors.New("failed to get sprint summary")
+	}
+
+	err = db.Model(&retroModels.Sprint{}).
+		Scopes(retroModels.SprintJoinSM).
+		Where("sprints.id = ?", sprintID).
+		Select("COUNT(*) as member_count, " +
+			"SUM(allocation_percent) as total_allocation, " +
+			"SUM(expectation_percent) as total_expectation, " +
+			"sum(vacations) as total_vacations," +
+			"0 as holidays").
+		Scan(&summary).Error
+
+	if err != nil {
+		utils.LogToSentry(err)
+		return summary, http.StatusInternalServerError, errors.New("failed to get sprint summary")
+	}
+
+	summary.TargetSP = utils.CalculateExpectedSP(*sprint.StartDate, *sprint.EndDate, summary.TotalVacations / float64(summary.MemberCount),
+		summary.TotalAllocation, summary.TotalExpectation, sprint.Retrospective.StoryPointPerWeek)
+
+	taskTypesSummary, status, err := service.GetSprintTaskTypesSummary(sprintID, retroID)
+
+	summary.TaskTypes = taskTypesSummary
+	if err != nil {
+		return summary, status, errors.New("failed to get sprint")
+	}
+
+	return summary, http.StatusOK, nil
+}
+
+// GetSprintTaskTypesSummary ...
+func (service SprintService) GetSprintTaskTypesSummary(sprintID string, retroID uint) (summary map[string]retroSerializers.SprintTaskTypeSummary, status int, err error) {
+	db := service.DB
+	var retro retroModels.Retrospective
+
+	err = db.Model(&retroModels.Retrospective{}).Where("id = ?", retroID).First(&retro).Error
+	if err != nil {
+		if err == gorm.ErrRecordNotFound {
+			return summary, http.StatusNotFound, errors.New("retrospective not found")
+		}
+		utils.LogToSentry(err)
+		return summary, http.StatusInternalServerError, errors.New("failed to get sprint summary")
+	}
+
+	taskTypes, err := tasktracker.GetTaskTypeMappings(retro.TaskProviderConfig)
+
+	if err != nil {
+		utils.LogToSentry(err)
+		return summary, http.StatusInternalServerError, errors.New("failed to get sprint summary")
+	}
+	summary = make(map[string]retroSerializers.SprintTaskTypeSummary)
+
+	for _, taskType := range tasktracker.TaskTypes {
+		summary[taskType], status, err = service.getSprintTaskTypeQuery(sprintID, taskTypes[taskType])
+		if err != nil {
+			return summary, status, err
+		}
+	}
+
+	return summary, http.StatusOK, nil
+}
+
+func (service SprintService) getSprintTaskTypeQuery(sprintID string, taskTypes string) (summary retroSerializers.SprintTaskTypeSummary, status int, err error) {
```

Comment:
Is the method returning data or query?

Reply:
It is returning data. I changed the name.